packages/valory/skills/termination_abci/rounds.py

```diff
@@ -173,6 +173,7 @@ class PostTerminationTxAbciApp(AbciApp[Event]):
     # the following is not needed, it is added to satisfy the round check
     # the TerminationRound when run it terminates the agent, so nothing can come after it
     transition_function = {TerminationRound: {Event.TERMINATE: TerminationRound}}
+    # TODO: ensure this gets considered in chaining too
```

Comment:
Yes we have this in the backlog. What we could do is ensure that the `cross_period_persisted_keys` defined in the `TerminationAbciApp` get also merged into the `cross_period_persisted_keys` of the "normal" AbciApp being executed. This would require small modification to the `AbciApp.add_termination()`. 
This would help us make sure that the `cross_period_persisted_keys` of the TerminationAbciApp get persisted properly, as there is no need to make the assumption that the `cross_period_persisted_keys` of the TerminationAbciApp are necessary anywhere else.